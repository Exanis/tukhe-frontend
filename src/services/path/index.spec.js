import {
    api,
    oauthUrl,
    getFullUrlPath,
    parseQueryString
} from "./index";

it('Should have correct urls', () => {
    expect(api('test')).toMatchSnapshot();
    expect(oauthUrl()).toMatchSnapshot();
});

it('Should construct correct path', () => {
    const locationHttps = {
        protocol: 'https:',
        hostname: 'test',
        pathname: '/test'
    };
    const locationHttp = {
        ...locationHttps,
        protocol: 'http:'
    };
    const locationPort = {
        ...locationHttps,
        port: '8080'
    };
    const locationPartialPathName = {
        ...locationHttps,
        pathname: 'test'
    };

    expect(getFullUrlPath(locationHttps)).toMatchSnapshot();
    expect(getFullUrlPath(locationHttp)).toMatchSnapshot();
    expect(getFullUrlPath(locationPort)).toMatchSnapshot();
    expect(getFullUrlPath(locationPartialPathName)).toMatchSnapshot();
});

it('Should correctly parse query string', () => {
    expect(parseQueryString()).toMatchSnapshot();
    expect(parseQueryString('test')).toMatchSnapshot();
    expect(parseQueryString('test&test2=val')).toMatchSnapshot();
});
