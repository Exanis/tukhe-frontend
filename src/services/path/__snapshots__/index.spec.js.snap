// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should construct correct path 1`] = `"https://test:443/test"`;

exports[`Should construct correct path 2`] = `"http://test:80/test"`;

exports[`Should construct correct path 3`] = `"https://test:8080/test"`;

exports[`Should construct correct path 4`] = `"https://test:443/test"`;

exports[`Should correctly parse query string 1`] = `Object {}`;

exports[`Should correctly parse query string 2`] = `
Object {
  "test": true,
}
`;

exports[`Should correctly parse query string 3`] = `
Object {
  "test": true,
  "test2": "val",
}
`;

exports[`Should have correct urls 1`] = `"undefinedtest"`;

exports[`Should have correct urls 2`] = `undefined`;
